### PR TITLE
mirroring rockstor os edits for issue 131

### PR DESCRIPTION
**Pull request by phillxnet**
I have hopefully made a few improvements but found that the howto effectively doesn't work as it installs the root as ext4 not btrfs. I have made a note of this at the top of the page so it may well be as well to publish so we don't frustrate other users of the howto in the mean time.

Sphinx v1.2.3 compiles without errors and the results are as expected in Chrome.

### CODE REVIEW

File: mdraid-mirror/boot_drive_howto.rst [hunk 1]
```diff
@@ -3,41 +3,46 @@
 Mirroring Rockstor OS using Linux Raid
 ======================================
 
+N.B. this howto is a work in progress and doesn't currently work for a Rockstor
```
Comment (schakrava) [inline mdraid-mirror/boot_drive_howto.rst]: If we allow ext4 for root filesystem along with btrfs(like it used to be prior to rockstor-core pr 1158, it does work barring the fact that there won't be any root Pool as such. So users won't be able to create Shares or put data on the root filesystem in Rockstor supported ways. But all Pool/Share/Snapshot features work on other drives that may be on the system.

Am I missing something here?

**Comment by phillxnet** [comment]
@schakrava I will try next week to extend this pr so that we get btrfs on root back even when installing on mdraid as your have written so far.

**Comment by phillxnet** [comment]
Or I can just build on top of this pr if it needs to be merged (ext4 warn is in commit of it's own so we can always reverse that if need be for the time being, ie if code in pr 1158 re ext4 ends up being reverted for time being also).

**Comment by schakrava** [comment]
Thanks @phillxnet. Please do leave a comment here when ready for review.

**Comment by phillxnet** [comment]
I am currently working on this pr to add btrfs / modifications. This involves a major edit so changed pr name accordingly.

**Comment by phillxnet** [comment]
Leaving for today: mostly done and workable but needs final proof read and more testing of resulting install. Will try and finalize tomorrow (UK).

**Comment by phillxnet** [comment]
@schakrava OK, I've finally done it.

Submitting as works for me. I have tried on several virtual machines and a couple of real machines and I think this method, although long, will fit those who require a custom mdraid system disks install.
There are probably ways to trim some steps but I like it as is as it plays safe with installer state and does little behind it's back which I think is best, especially with a complex install such as this.

"make html" completes without errors using Sphinx v1.2.3 and the results are as expected in Chrome and Firefox.

Ready for review and Fixes #131

**Comment by schakrava** [comment]
Really impressive and very thorough @phillxnet !

**Comment by phillxnet** [comment]
@schakrava Thanks; we managed to pull it off between us in the end.

**Pull request merged**